TestProject/ComplexTests.cs.expected

```diff
@@ -0,0 +1,165 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using NUnit.Framework;
+using Rhino.Mocks;
+using TestProject.TestClasses;
+using TestProject.TestInterfaces;
+using Moq;
+using MockRepository = Rhino.Mocks.MockRepository;
+
+namespace TestProject
+{
+  [TestFixture]
+  public class ComplexTests
+  {
+    private ITestInterface _memberCaller;
+    private Order _order1;
+
+    private Customer _customer1;
+    private DomainObjectMockEventReceiver _order1MockEventReceiver;
+    private DomainObjectMockEventReceiver _customer1MockEventReceiver;
+    private ClientTransaction _subTransaction;
+    private Mock<ITestInterface> _serviceStub;
+    private Mock<ITypeInformation> _typeInformationForResourceResolutionStub;
+    private Mock<ITypeInformation> _typeInformationStub;
+    private Mock<IPropertyInformation> _propertyInformationStub;
+
+    [SetUp]
+    public void SetUp ()
+    {
+      _memberCaller = new TestClass();
+      _serviceStub = new Mock<ITestInterface>();
+      _typeInformationStub = new Mock<ITypeInformation>();
+      _typeInformationForResourceResolutionStub = new Mock<ITypeInformation>();
+      _propertyInformationStub = new Mock<IPropertyInformation>();
+    }
+
+    [Test]
+    public void InitRecursive ()
+    {
+      Page namingContainer = new Page();
+      var parentControlMock = new Mock<IControl>() { CallBase = true };
+      var childControlMock = new Mock<IControl>() { CallBase = true };
+      var sequence1 = new MockSequence();
+      childControlMock.InSequence (sequence1).Setup (mock => PrivateInvoke.InvokeNonPublicMethod (mock, "OnInit", EventArgs.Empty)).Verifiable();
+      parentControlMock.InSequence (sequence1).Setup (mock => PrivateInvoke.InvokeNonPublicMethod (mock, "OnInit", EventArgs.Empty)).Verifiable();
+
+      namingContainer.Controls.Add (parentControlMock.Object);
+      parentControlMock.Object.Controls.Add (childControlMock.Object);
+      _memberCaller.InitRecursive (parentControlMock.Object, namingContainer);
+
+      parentControlMock.Verify();
+
+      childControlMock.Verify();
+    }
+
+    [Test]
+    public void SubCommit_OfDeletedObject_DoesNotRaiseDeletedEvent ()
+    {
+      using (_subTransaction.EnterDiscardingScope())
+      {
+        ITestInterface domainObject = new TestClass();
+
+        var extensionMock = new Mock<IClientTransactionExtension> (MockBehavior.Strict);
+        extensionMock.Setup (stub => stub.Key).Returns ("Mock");
+        _subTransaction.Extensions.Add (extensionMock.ToString());
+        try
+        {
+
```

Comment:
Deleting `BackToRecord` left the blank line behind. The rule would be if there's no more code between the block start and the removed/updated code, remove the blank lines.